WebService/v1/Controllers/SolutionSettings.cs

```diff
@@ -41,24 +44,45 @@ public SolutionSettingsController(IStorage storage)
         [HttpPut("solution-settings/logo")]
         public async Task SetLogoAsync()
         {
-            var bytes = new byte[this.Request.Body.Length];
-            this.Request.Body.Read(bytes, 0, (int) this.Request.Body.Length);
+            MemoryStream memoryStream = new MemoryStream();
+            this.Request.Body.CopyTo(memoryStream);
+            byte[] bytes = memoryStream.ToArray();
 
             var model = new Logo
             {
-                Image = Convert.ToBase64String(bytes),
-                Type = this.Request.Headers["content-type"]
+                IsDefault = false
             };
 
+            if (bytes.Length > 0)
+            {
+                model.SetImageFromBytes(bytes);
+                model.Type = this.Request.ContentType;
+            }
+
+            if (this.Request.Headers[Logo.NAME_HEADER] != StringValues.Empty)
+            {
+                model.Name = this.Request.Headers[Logo.NAME_HEADER];
+            }
+        
             var response = await this.storage.SetLogoAsync(model);
             this.SetImageResponse(response);
         }
 
         private void SetImageResponse(Logo model)
         {
-            var bytes = Convert.FromBase64String(model.Image);
-            this.Response.Headers.Add("content-type", model.Type);
-            this.Response.Body.Write(bytes, 0, bytes.Length);
+            if(model.Name != null)
+            {
+                this.Response.Headers.Add(Logo.NAME_HEADER, model.Name);
```

Comment:
If ```model.Name``` is null ```Logo.NAME_HEADER``` will be missing from the headers, will that be a problem?

Reply:
No that won't be a problem